src/components/table-hoc/TableWithPagination.tsx

```diff
@@ -0,0 +1,116 @@
+import * as React from 'react';
+import {keys} from 'ts-transformer-keys/index';
+import * as _ from 'underscore';
+import {IReactVaporState, IReduxActionsPayload} from '../../ReactVapor';
+import {callIfDefined} from '../../utils/FalsyValuesUtils';
+import {IReduxAction, ReduxConnect} from '../../utils/ReduxUtils';
+import {turnOffLoading} from '../loading/LoadingActions';
+import {INavigationChildrenProps, INavigationOwnProps} from '../navigation/Navigation';
+import {NavigationConnected} from '../navigation/NavigationConnected';
+import {TableWithPaginationActions} from './actions/TablePaginationActions';
+import {ITableHOCOwnProps} from './TableHOC';
+
+export interface ITableWithPaginationConfig {
+    sliceData?: (data: any[], startingIndex: number, endingIndex: number) => any[];
+    isServer?: boolean;
+}
+
+export interface ITableWithPaginationOwnProps {}
+
+export interface ITableWithPaginationStateProps {
+    totalEntries: number;
+    totalPages: number;
+    pageNb: number;
+    perPage: number;
+}
+
+export interface ITableWithPaginationDispatchProps {
+    onMount: () => void;
+    onUnmount: () => void;
+}
+
+export interface ITableWithPaginationProps extends Partial<ITableWithPaginationStateProps>,
+    Partial<ITableWithPaginationDispatchProps> {}
+
+const TableWithPaginationPropsToOmit = keys<ITableWithPaginationStateProps>();
+const TableWithPaginationConfigToOmit = keys<ITableWithPaginationConfig>();
+
+const defaultSliceData = (data: any[], startingIndex: number, endingIndex: number) => data.slice(startingIndex, endingIndex);
+
+export const tableWithPagination = (config: ITableWithPaginationConfig & Partial<INavigationOwnProps & INavigationChildrenProps> = {}) => (Component: (React.ComponentClass<ITableHOCOwnProps> | React.StatelessComponent<ITableHOCOwnProps>)): React.ComponentClass<ITableWithPaginationProps & React.HTMLAttributes<HTMLTableElement>> => {
+    const mapStateToProps = (state: IReactVaporState, ownProps: ITableHOCOwnProps & ITableWithPaginationOwnProps): ITableWithPaginationStateProps | ITableHOCOwnProps => {
+        const paginationState = _.findWhere(state.paginationComposite, {id: `pagination-${ownProps.id}`});

```

Comment:
imho `pagination-${ownProps.id}` and `loading-${ownProps.id}` should be generated by a method similar to getTableChildComponentId in TableUtils.ts  

we had bugs related to this id logic in react-vapor at some point in the past, we had forgotten to change it in some places and it broke something at run time.

Reply:
I made something with this in mind in #908